pgrx/src/spi/client.rs

```diff
@@ -5,22 +5,26 @@ use std::ptr::NonNull;
 use crate::pg_sys::{self, PgOid};
 use crate::spi::{PreparedStatement, Query, Spi, SpiCursor, SpiError, SpiResult, SpiTupleTable};
 
-// TODO: should `'conn` be invariant?
-pub struct SpiClient<'conn> {
-    __marker: PhantomData<&'conn SpiConnection>,
-}
+pub struct SpiClient;
+
+impl SpiClient {
+    /// Connect to Postgres' SPI system
+    pub(super) fn connect() -> SpiResult<Self> {
+        // connect to SPI
+        //
+        // SPI_connect() is documented as being able to return SPI_ERROR_CONNECT, so we have to
+        // assume it could.  The truth seems to be that it never actually does.  The one user
+        // of SpiConnection::connect() returns `spi::Result` anyways, so it's no big deal
+        Spi::check_status(unsafe { pg_sys::SPI_connect() })?;
+        Ok(SpiClient)
+    }
 
-impl<'conn> SpiClient<'conn> {
     /// Prepares a statement that is valid for the lifetime of the client
     ///
     /// # Panics
     ///
     /// This function will panic if the supplied `query` string contained a NULL byte
-    pub fn prepare(
-        &self,
-        query: &str,
-        args: Option<Vec<PgOid>>,
-    ) -> SpiResult<PreparedStatement<'conn>> {
+    pub fn prepare(&self, query: &str, args: Option<Vec<PgOid>>) -> SpiResult<PreparedStatement> {
```

Comment:
I'm surprised this is valid?

Reply:
How so?

Do you mean it might need to be 
```rust
fn prepare<'client>(&'client self, ...) -> SpiResult<PreparedStatement<'client>> {}
```

Reply:
I expected Rust to ask for `SpiResult<PreparedStatement<'_>> at least.